File: Java/443.string-compression.java
class Solution {
	public int compress(char[] chars) {
		int j = 0, curRun = 1, n = chars.length;
		for (int i=1;i<=n;i++,curRun++) {
			if (i < n && chars[i] == chars[i-1]) continue;           
			chars[j++] = chars[i-1];
			if (curRun != 1) {
				String s = Integer.toString(curRun);
				for (int k=j;k<j+s.length();k++) chars[k] = s.charAt(k-j);
				j += s.length();  
			}
			curRun = 0;    
		}
		return j;
	}
}

File: Easy/443.string-compression.js
/**
 * @param {character[]} chars
 * @return {number}
 */
var compress = function (chars) {
  chars.sort();
  if (chars.length === 0) {
    return 0;
  }

  if (chars.length === 1) {
    return 1;
  }


  let currentChar = chars[0];
  let count = 0;
  let result = [];

  while (chars.length) {
    let char = chars.shift();
    if (char == currentChar) {
      count++;
    } else {
      if (count == 1) {
        result.push(char);
      } else {
        let length = count.toString().split("");
        result.push(currentChar);
        result.concat(length);
      }
      currentChar = char;
      count = 1;
    }
  }
  
  if (count == 1) {
    result.push(currentChar);
  } else {
    let length = count.toString().split("");
    result.push(currentChar);
    result.concat(length);
  }

  return result;
};